tests/std/tests/P0645R10_text_formatting_args/test.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <concepts>
+#include <cstddef>
+#include <format>
+#include <memory>
+#include <string_view>
+#include <type_traits>
+#include <variant>
+
+using namespace std;
+
+#pragma warning(push)

```

Comment:
It's silly to push the warning state when we're not going to pop until the end of the TU.